include/operators/binary_arithmetic.hpp

```diff
@@ -180,10 +182,15 @@ BigInt BigInt::operator*(const BigInt& num) const {
         num2_high = smaller.substr(0, half_length);
         num2_low = smaller.substr(half_length);
 
-        strip_leading_zeroes(num1_high.value);
-        strip_leading_zeroes(num1_low.value);
-        strip_leading_zeroes(num2_high.value);
-        strip_leading_zeroes(num2_low.value);
+        std::thread t1(strip_leading_zeroes, std::ref(num1_high.value));
+        std::thread t2(strip_leading_zeroes, std::ref(num1_low.value));
+        std::thread t3(strip_leading_zeroes, std::ref(num2_high.value));
+        std::thread t4(strip_leading_zeroes, std::ref(num2_low.value));
```

Comment:
The threads `t1`, `t2`, ... aren't very descriptive and should be renamed to `thread1`, `thread2`, ...

Reply:
Added the fix :)